src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
+        }
+    }
+}
+
+void EnEgol_SetupStop(EnEgol* this) {
+    if (!this->isRetreating) {
+        EnEgol_SetAnimation(this, EYEGORE_ANIM_LASER);
+    } else {
+        EnEgol_SetAnimation(this, EYEGORE_ANIM_STUN_END);
+    }
+    this->actionState = EYEGORE_ACTION_STOP;
+    this->actionFunc = EnEgol_Stop;
+}
+
+void EnEgol_Stop(EnEgol* this, PlayState* play) {
+    s32 isHome = false;
+    f32 curFrame = this->skelAnime.curFrame;
+    s16 angleToHome;
+
+    EnEgol_FootstepEffects(this, play, 28.0f, 13.0f);
+    if (this->isRetreating) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 5, 0xBB8, 5);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        angleToHome = this->actor.world.rot.y - this->actor.home.rot.y;
+    }
+    if (!this->isRetreating) {
+        isHome = true;
+    } else if (ABS_ALT(angleToHome) < 0x64) {
+        this->actor.world.rot.y = this->actor.shape.rot.y = this->actor.home.rot.y;
+        isHome = true;
+    }
+    if (isHome && (curFrame >= this->animEndFrame)) {
+        this->isRetreating ^= 1;
+        if (!this->isRetreating) {
+            EnEgol_SetAnimation(this, EYEGORE_ANIM_SIT);
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_SIT);
+            this->laserCount = 0;
+            this->actionFunc = EnEgol_Wait;
+        } else {
+            EnEgol_SetupRetreat(this);
+        }
+    }
+}
+
+void EnEgol_SetupSlam(EnEgol* this) {
+    EnEgol_SetAnimation(this, EYEGORE_ANIM_SLAM);
+    this->actionState = EYEGORE_ACTION_SLAM;
+    this->actionFunc = EnEgol_Slam;
+}
+
+void EnEgol_Slam(EnEgol* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    Math_SmoothStepToS(&this->headRot, 0, 1, 0x3E8, 0);
+    if (Animation_OnFrame(&this->skelAnime, 17.0f)) {
+        s32 i;
+        s32 spawnCount;
+        Player* player = GET_PLAYER(play);
+        Vec3f spawnPos;
+        f32 dyToPlayer = fabsf(player->actor.world.pos.y - this->actor.world.pos.y);
+        s32 quakeSize;
+
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_EXPLOSION);
+        quakeSize = 4 - (s32)(dyToPlayer * 0.02f);
+        if (quakeSize > 4) {
+            quakeSize = 4;
+        } else if (quakeSize < 1) {
+            quakeSize = 1;
+        }
+        func_800BC848(&this->actor, play, quakeSize, 2);
+        if (this->actor.floorBgId == BGCHECK_SCENE) {
+            Math_Vec3f_Copy(&spawnPos, &this->actor.world.pos);
+            spawnPos.x += Math_SinS(this->actor.world.rot.y) * 60.0f;
+            spawnPos.y = this->actor.floorHeight;
+            spawnPos.z += Math_CosS(this->actor.world.rot.y) * 60.0f;
+            spawnCount = Rand_S16Offset(1, 3);
+            for (i = 0; i < spawnCount; i++) {
+                Actor_Spawn(&play->actorCtx, play, ACTOR_EN_ESTONE, spawnPos.x, spawnPos.y, spawnPos.z, 0,
+                            this->actor.world.rot.y, 0, ENESTONE_LARGE);
```

Comment:
Usually params enums get `TYPE` to not confuse them with other enums related to that actor.